### PR TITLE
Windows: apply skip taskbar state when taskbar is restarted

### PR DESCRIPTION
**Pull request by amrbashir**
- [x] Tested on all platforms changed
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x] Created or updated an example program if it would help users understand this functionality
- [x] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

**Comment by amrbashir** [comment]
seems like rust nightly added a `#[must_use]` on `Box::from_raw` and it is causing build/lint errors

**Comment by maroider** [comment]
> when taskbar is restarted

When and how does this happen exactly? Is this only an issue if you kill `explorer.exe` and start it back up again?

> seems like rust nightly added a #[must_use] on Box::from_raw and it is causing build/lint errors

A fix for that should have landed alongside #2381.

**Review by msiglreith — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ And please only add new entries to the top of this list, right below the `# Unre
 
 # Unreleased
 
+- On Windows, apply skip taskbar state when taskbar is restarted.
```
Comment (msiglreith) [inline CHANGELOG.md:11]: could you remove the changelog entry here? `set_skip_taskbar` will make its first appearance in the upcoming release

Comment (amrbashir) [reply CHANGELOG.md]: okay

**Pull request merged**